From patches Mon Sep 17 00:00:00 2001
From: liguang <lig.fnst@cn.fujitsu.com>
To: afaerber@suse.de, qemu-devel@nongnu.org
Cc: liguang <lig.fnst@cn.fujitsu.com>
Date: Wed, 10 Apr 2013 16:02:29 +0800
Message-id: 1365580949-32467-2-git-send-email-lig.fnst@cn.fujitsu.com
Subject: [PATCH 2/2] target-i386: fix 2 malformed indents

Signed-off-by: liguang <lig.fnst@cn.fujitsu.com>
---
 target-i386/cpu.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/target-i386/cpu.c b/target-i386/cpu.c
index 6e13053..9806a60 100644
--- a/target-i386/cpu.c
+++ b/target-i386/cpu.c
@@ -2108,7 +2108,7 @@ static void x86_cpu_realizefn(DeviceState *dev, Error **errp)
         env->cpuid_vendor3 == CPUID_VENDOR_AMD_3) {
         env->cpuid_ext2_features &= ~CPUID_EXT2_AMD_ALIASES;
         env->cpuid_ext2_features |= (env->cpuid_features
-           & CPUID_EXT2_AMD_ALIASES);
+                                     & CPUID_EXT2_AMD_ALIASES);
     }
 
     if (!kvm_enabled()) {
@@ -2116,7 +2116,7 @@ static void x86_cpu_realizefn(DeviceState *dev, Error **errp)
         env->cpuid_ext_features &= TCG_EXT_FEATURES;
         env->cpuid_ext2_features &= (TCG_EXT2_FEATURES
 #ifdef TARGET_X86_64
-            | CPUID_EXT2_SYSCALL | CPUID_EXT2_LM
+                                     | CPUID_EXT2_SYSCALL | CPUID_EXT2_LM
 #endif
             );
         env->cpuid_ext3_features &= TCG_EXT3_FEATURES;
-- 
1.7.2.5


